src/legacy/core_plugins/kibana/public/discover/directives/histogram.tsx

```diff
@@ -44,22 +45,45 @@ import {
 
 import { i18n } from '@kbn/i18n';
 
-import { getChartTheme } from 'ui/elastic_charts';
 import chrome from 'ui/chrome';
 // @ts-ignore: path dynamic for kibana
 import { timezoneProvider } from 'ui/vis/lib/timezone';
+import { EuiChartThemeType } from '@elastic/eui/src/themes/charts/themes';
+import { Subscription } from 'rxjs';
 
 export interface DiscoverHistogramProps {
   chartData: any;
   timefilterUpdateHandler: (ranges: { from: number; to: number }) => void;
 }
 
-export class DiscoverHistogram extends Component<DiscoverHistogramProps> {
+interface DiscoverHistogramState {
+  chartsTheme: EuiChartThemeType['theme'];
+}
+
+export class DiscoverHistogram extends Component<DiscoverHistogramProps, DiscoverHistogramState> {
   public static propTypes = {
     chartData: PropTypes.object,
     timefilterUpdateHandler: PropTypes.func,
   };
 
+  private subscription?: Subscription;
+  public state = {
+    chartsTheme: npStart.plugins.eui_utils.getChartsThemeDefault(),
+  };
+
+  componentDidMount() {
+    this.subscription = npStart.plugins.eui_utils
+      .getChartsTheme$()
+      .subscribe(chartsTheme => this.setState({ chartsTheme }));
+  }
+
+  componentDidUnmount() {
```

Comment:
should be [`componentWillUnmount`](https://reactjs.org/docs/react-component.html#componentwillunmount)

```suggestion
  componentWillUnmount() {
```

Reply:
Updated